Prolangle/Pages/Home.razor

```diff
@@ -1,7 +1,313 @@
+		{
+			<TableRow>
+				<TableRowHeader Style="white-space: nowrap; vertical-align: middle">@language.Name</TableRowHeader>
+				<TableRowCell>
+					<MatchDisplay Match="TypingMatch(language)">
+						<LanguageMetadatum Overlap="TypingOverlap(language)" Value="language.Typing" Exclude="TypeSystem.None"/>
+					</MatchDisplay>
+				</TableRowCell>
+				<TableRowCell>
+					<MatchDisplay Match="CompiledMatch(language)">
+						@(language.Compiled ? "Compiled" : "Interpreted")
+					</MatchDisplay>
+				</TableRowCell>
+				<TableRowCell>
+					<MatchDisplay Match="GarbageCollectedMatch(language)">
+						<LanguageMemoryManagement MemoryManagement="language.MemoryManagement"/>
+					</MatchDisplay>
+				</TableRowCell>
+				<TableRowCell>
+					<MatchDisplay Match="SyntaxStyleMatch(language)">
+						<LanguageSyntaxStyle SyntaxStyle="language.SyntaxStyle"/>
+					</MatchDisplay>
+				</TableRowCell>
+				<TableRowCell>
+					<MatchDisplay Match="ApplicationsMatch(language)">
+						<LanguageMetadatum Overlap="ApplicationsOverlap(language)" Value="language.KnownForBuilding" Exclude="Applications.None"/>
+					</MatchDisplay>
+				</TableRowCell>
+				<TableRowCell>
+					<MatchDisplay Match="ParadigmsMatch(language)">
+						<LanguageMetadatum Overlap="ParadigmsOverlap(language)" Value="language.Paradigms" Exclude="Paradigms.None"/>
+					</MatchDisplay>
+				</TableRowCell>
+				<TableRowCell>
+					<MatchDisplay Match="TiobeRatingMatch(language)">
+						@(language.TiobeRating?.ToString() ?? "None")
+						@DirectionToString(TiobeRatingDirection(language))
+					</MatchDisplay>
+				</TableRowCell>
+				<TableRowCell>
+					<MatchDisplay Match="AppearanceYearMatch(language)">
+						@language.AppearanceYear
+						@DirectionToString(AppearanceYearDirection(language))
+					</MatchDisplay>
+				</TableRowCell>
+			</TableRow>
+		}
+	</TableBody>
+</Table>
+
+@if (won)
+{
+	<Heading Size="HeadingSize.Is2">You won!</Heading>
+}
+else
+{
+	<LanguageSelection AvailableLanguages="@AvailableLanguages" OnLanguageSelected="@OnLanguageSelected"/>
+}
+
+@code {
+	private List<ILanguage> AvailableLanguages => LanguagesProvider!.Languages.Where(l => !RevealedLanguages.Contains(l)).ToList();
+	private List<ILanguage> RevealedLanguages { get; } = [];
+	private ILanguage TargetLanguage => LanguagesProvider!.Languages.Single(l => l.Id == GuessGame!.TargetLanguageId);
+	private bool won;
+
+	private static string DirectionToString(DirectionIndicator direction)
+	{
+		return direction switch
+		{
+			DirectionIndicator.Equal => "",
+			DirectionIndicator.Up => "\u25b2",
+			DirectionIndicator.Down => "\u25bc",
+			_ => throw new ArgumentOutOfRangeException(nameof(direction), direction, null)
+		};
+	}
+
+	private void OnLanguageSelected(ILanguage language)
+	{
+		RevealedLanguages.Add(language);
+
+		var result = GuessGame!.TargetLanguageId == language.Id;
+
+		if (result)
+			won = true;
+	}
+
+	private TypeSystem TypingOverlap(ILanguage language)
```

Comment:
Ok, sure. Are you opposed to moving the whole game to its own component? That way the `Home.razor` would only include that component for now, but would open up the possibility for more separation.

Reply:
I did it and I think its more clean this way. The `Home.razor` keeps track of the target language and the revealed languages and the `MetadataComparisonTable` takes care of the presentation.

Reply:
That's better, but what I meant was that the comparison logic isn't really a UI concern, IMHO.

Reply:
The logic is moved to a service too, in case you didn't notice. The new table component uses the service to do the comparisons and truly only cares about representing it.

Reply:
Yeah, never mind. Much nicer.